app/services/payment_providers/stripe/payments/create_service.rb

```diff
@@ -216,13 +229,14 @@ def error_on_requires_action?
           invoice.customer.country != "IN"
         end
 
-        def prepare_failed_result(error, reraise: false, payable_payment_status: :failed)
+        def prepare_failed_result(error, reraise: false, payment_status: :failed, payable_payment_status: :failed, should_retry: false)
           result.error_message = error.message
           result.error_code = error.code
           result.reraise = reraise
+          result.should_retry = should_retry
 
           # stripe may return us a Stripe::CardError error if payment_intent was created, but it's processing failed, in this case error would contain payment_intent id
-          payment.update!(status: :failed, payable_payment_status:, provider_payment_id: error.error&.payment_intent&.id)
+          payment.update!(status: payment_status, payable_payment_status:, provider_payment_id: error.error&.payment_intent&.id, error_code: error.code)
```

Comment:
Notice that we save the error code for all errors, not just the error I'm interested in 👌